### ISSUE TITLE
fix boundary vertices without introducing additional vertices

### ISSUE THREAD

**Issue by xiaolong7** [huxingyi/isotropicremesher#4]
Hi, this is not an issue of the code but a question that I hope you could help to point me to the right direction. I want to have a final mesh that has exactly **the same boundary vertices as in the original mesh** without any additional or missing vertices on the boundary. I am wondering if it is possible to achieve this by just making edits on few lines of the code? It would be great if that's possible and if you could point me to the part of code that needs to be edited. Thank you for your time and help!

**Comment by huxingyi** [comment]
Add the following check

https://github.com/huxingyi/isotropicremesher/blob/d4674ce63288d7a55a70735181049895aebc3bf7/src/isotropicremesher.cpp#L195

before breaking the edge

https://github.com/huxingyi/isotropicremesher/blob/d4674ce63288d7a55a70735181049895aebc3bf7/src/isotropicremesher.cpp#L164

**Comment by xiaolong7** [comment]
> Add the following check
> 
> https://github.com/huxingyi/isotropicremesher/blob/d4674ce63288d7a55a70735181049895aebc3bf7/src/isotropicremesher.cpp#L195
> 
> before breaking the edge
> 
> https://github.com/huxingyi/isotropicremesher/blob/d4674ce63288d7a55a70735181049895aebc3bf7/src/isotropicremesher.cpp#L164

Thank you for your suggestion! 
I have made the following change according to my understanding of your suggestion, but there are still some additional vertices on the boundary of the final mesh. Does the change I made follow your suggestion?
![image](https://user-images.githubusercontent.com/104316304/208716458-03730180-ef49-4112-81b8-87a1c646ae6b.png)

**Comment by xiaolong7** [comment]
It works after I changed the condition to 
`if (!halfedge->startVertex->_isBoundary && !nextHalfedge->startVertex->_isBoundary) `.
Thank you!

**Comment by huxingyi** [comment]
You are right, the boundary vertices are not featured before the long edge splits, so should use the boundary check directly.

**Comment by mo1shang** [comment]
Is there a way to preserve the sharp features of the model? The cube-sphere.obj model in the test directory has sharp features.

**Comment by Andy3531** [comment]
flips may occur
